skema/program_analysis/CAST/matlab/matlab_to_cast.py

```diff
@@ -796,380 +529,206 @@ def visit_math_expression(self, node):
+                    source_code_data_type=["matlab", MATLAB_VERSION, "unknown"],
+                    source_refs=[self.node_helper.get_source_ref(cell_node)]
+                )
+                return get_operator("in", [identifier, operand], source_refs)
+            # single case argument
+            nodes = get_children_by_types(case_node, case_node_types)
+            operand = [self.visit(node) for node in nodes][0]
+            return get_operator("==", [identifier, operand], source_refs)
 
-        # The left side is equivilent to a call gromet "get", so we will first pass the left side to the get visitor
-        # Then we can easily convert this to a "set" call by modifying the fields and then appending the right side to the function arguments
-        cast_call = self._visit_get(left)
-        cast_call.func = self.get_gromet_function_node("set")
-        cast_call.arguments.append(self.visit(right))
+        def get_case_body(case_node):
+            """ return the instruction block for the case """
+            block = get_first_child_by_type(case_node, "block")
+            if block:
+                return [self.visit(c) for c in get_keyword_children(block)]
+            return None
+            
+        def get_model_if(case_node, identifier):
+            """ return conditional logic representing the case """
+            return ModelIf(
+                expr = get_case_expression(case_node, identifier),
+                body = get_case_body(case_node),
+                source_refs=[self.node_helper.get_source_ref(case_node)]
+            )
+        
+        # switch statement identifier
+        identifier = self.visit(get_first_child_by_type(node, "identifier"))
+        
+        # n case clauses as 'if then' nodes
+        case_nodes = get_children_by_types(node, ["case_clause"])
+        model_ifs = [get_model_if(node, identifier) for node in case_nodes]
+        for i, model_if in enumerate(model_ifs[1:]):
+            model_ifs[i].orelse = [model_if]
 
-        return cast_call
+        # otherwise clause as 'else' node after last 'if then' node
+        otherwise_clause = get_first_child_by_type(node, "otherwise_clause")
+        if otherwise_clause:
+            block = get_first_child_by_type(otherwise_clause, "block")
+            if block:
+                last = model_ifs[len(model_ifs)-1]
+                last.orelse = [self.visit(c) for c in get_keyword_children(block)]
+        return model_ifs[0]
 
-    def _visit_while(self, node) -> Loop:
-        """Custom visitor for while loop. Returns a Loop object"""
+    # into a CAST-supported loop type.
+    def visit_while_statement(self, node) -> Loop:
+        """ Translate MATLAB while_loop syntax node into CAST Loop node """
         """
-        Node structure
-        Do while
-        (do_loop_statement)
-            (while_statement)
-                (parenthesized_expression)
-                    (...) ...
-            (body) ...
+        SOURCE:
```

Comment:
I think at some point in the future it makes sense to remove these comments that have the code and syntax tree, or at least move them somewhere out of the way. This way the code is easier to read through.
However I think for now it's fine since I imagine it helps out with development.

Reply:
Those were Tree-sitter output that I use for designing CAST structures for unimplemented syntax keywords.   I have removed them, they are easy to regenerate as I need them.